app/controllers/users_controller.rb

```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by!(username: params.fetch(:id))
```

Comment:
It's usually best practice to keep the name of the parameter the same as the column you want to look in, when applicable. Since there is only one route for users `resource`

https://github.com/AppdevBoothOrg/industrial-chat-application-1/blob/bde2001c59544c7bfaf0307e1936a16b66b68841/config/routes.rb#L17

You could drop this in favor of something more explicit like

```rb
get "/users/:username" => "users#show"
```

then you can fetch `:username` instead of `:id`. What do you think?